packages/server-core/src/projects/project/project-helper.ts

```diff
@@ -778,63 +782,87 @@ export const findBuilderTags = async (): Promise<Array<ProjectBuilderTagsType>>
+            pushedAt: imageDetails.imagePushedAt!.toJSON()
+          }
+        })
+    } catch (err) {
+      logger.error('Failure to get public ECR images')
+      logger.error('Command that was sent', result)
+      logger.error(err)
+      return []
+    }
   } else if (privateECRExec) {
+    const awsCredentials = `[default]\naws_access_key_id=${config.aws.eks.accessKeyId}\naws_secret_access_key=${config.aws.eks.secretAccessKey}\n[role]\nrole_arn = ${config.aws.eks.roleArn}\nsource_profile = default`
+
+    if (!fs.existsSync(awsPath)) fs.mkdirSync(awsPath, { recursive: true })
+    fs.writeFileSync(credentialsPath, Buffer.from(awsCredentials))
+
     const ecr = new ECRClient({
-      credentials: {
-        accessKeyId: config.aws.eks.accessKeyId,
-        secretAccessKey: config.aws.eks.secretAccessKey
-      },
+      credentials: fromIni({
+        profile: config.aws.eks.roleArn ? 'role' : 'default',
+        filepath: credentialsPath
+      }),
       region: privateECRExec[1]
     })
     const command = {
       repositoryName: privateECRExec[2]
     }
-    const result = new DescribeImagesCommand(command)
-    const response = await ecr.send(result)
-    if (!response || !response.imageDetails) return []
-    return response.imageDetails
-      .filter(
-        (imageDetails) => imageDetails.imageTags && imageDetails.imageTags.length > 0 && imageDetails.imagePushedAt
-      )
-      .sort((a, b) => b.imagePushedAt!.getTime() - a.imagePushedAt!.getTime())
-      .map((imageDetails) => {
-        const tag = imageDetails.imageTags!.find((tag) => !/latest/.test(tag))!
-        const tagSplit = tag ? tag.split('_') : ''
-        return {
-          tag,
-          commitSHA: tagSplit.length === 1 ? tagSplit[0] : tagSplit[1],
-          engineVersion: tagSplit.length === 1 ? 'unknown' : tagSplit[0],
-          pushedAt: imageDetails.imagePushedAt!.toJSON()
-        }
-      })
+    const result = new DescribePrivateImagesCommand(command)
+    try {
+      const response = await ecr.send(result)
+      if (!response || !response.imageDetails) return []
+      return response.imageDetails
+        .filter(
+          (imageDetails) => imageDetails.imageTags && imageDetails.imageTags.length > 0 && imageDetails.imagePushedAt
+        )
+        .sort((a, b) => b.imagePushedAt!.getTime() - a.imagePushedAt!.getTime())
+        .map((imageDetails) => {
+          const tag = imageDetails.imageTags!.find((tag) => !/latest/.test(tag))!
+          const tagSplit = tag ? tag.split('_') : ''
+          return {
+            tag,
+            commitSHA: tagSplit.length === 1 ? tagSplit[0] : tagSplit[1],
+            engineVersion: tagSplit.length === 1 ? 'unknown' : tagSplit[0],
+            pushedAt: imageDetails.imagePushedAt!.toJSON()
+          }
+        })
+    } catch (err) {
+      logger.error('Failure to get private ECR images')
+      logger.error('Command that was sent %o', result)
+      logger.error(err)
```

Comment:
Do we have a way to suppress these log statements on certain deployments? For instance AWS error messages can contain sensitive data, so I doubt we want these in the console on production.

Reply:
I don't think we have that right now.